### DISCUSSION TITLE
YouTube imagemagick command

### DISCUSSION THREAD

**Discussion by eddievalentin07** [ImageMagick/ImageMagick#6787] [Digital Image Processing] [answered]
im currently using this command in imagemagick

C:\ImageMagick\convert.exe -background black -fill white -font Courier -pointsize 72 -gravity center label:"Text Here" -extent 1920x1080 image.png

i found ffmpeg youtube command for youtube recomendation settings with this link https://trac.ffmpeg.org/wiki/Encode/YouTube

and im using the still image command 

ffmpeg -loop 1 -framerate 2 -i input.png -i audio.m4a -c:v libx264 -preset medium -tune stillimage -crf 18 -c:a copy -shortest -pix_fmt yuv420p output.mkv

Now, I need to find Imagemagick YouTube Recommendations. Can someone give me the exact command that match with YouTube?

**Comment by snibgo** [comment]
I don't understand the question.

The IM command creates a PNG with a single image. The ffmpeg command makes a video from a single image and an audio file. That command works for me, creating a video that Windows Media Player can play.

If YouTube can't accept the video, or mangles it, that would be a problem for YouTube.

In your other thread https://github.com/ImageMagick/ImageMagick/discussions/6773 I linked to a YouTube page that says:
> make sure that you’re using one of the following formats:

.. and MKV is not among the listed formats.

**Reply by eddievalentin07** [reply]
my question is is there a ffmpeg YouTube command line/parameter. i apologize if im confusing you

**Reply by fmw42** [reply]
questions about ffmpeg should be addressed that forum.  Are you asking is there an Imagemagick command equivalent to the ffmpeg command?

**Reply by eddievalentin07** [reply]
something like that but i need to know what imagemagick command that match and be equivalent to YouTube. know any command? i already found the one in ffmpeg for youtube settings. I tried asking other websites please help

**Comment by snibgo** [comment]
ImageMagick has no options that are specific to YouTube.

**Reply by eddievalentin07** [reply]
damn. Well do you know how to convert a 1080p full hd black background with text in the middle 32 bit? please?

**Reply by fmw42** [reply]
Can you explain what you want for a result and what you have to start?  Can you provide or point to an example?

**Reply by eddievalentin07** [reply]
Im currently using this

C:\ImageMagick\convert.exe -background black -fill white -font Courier -pointsize 72 -gravity center label:"Text Here" -extent 1920x1080 image.png

but when i right click and check properties it's 8 bit depth and i want it to be at least 32 bit depth.

**Reply by fmw42** [reply]
Preface your output with PNG32: such as PNG32:image.png

**Reply by eddievalentin07** [reply]
> Can you explain what you want for a result and what you have to start?  Can you provide or point to an example?
> 
> 

btw do you any full hd command with black background and text in the center? I apologize if im confusing you guys?

**Reply by fmw42** [reply]
Sorry, I do not understand your question. Please elaborate.

**Reply by eddievalentin07** [reply]
> Preface your output with PNG32: such as PNG32:image.png
> 
> 
> 
> 

Thank You.

**Reply by eddievalentin07** [reply]
> Sorry, I do not understand your question. Please elaborate.

One more thing i convert black background with text in center but the thing is it doesn't sound too good and it turns out to be 8 bit depth. How to convert black screen/background with text in the center full hd 1080p please. Any command examples?

**Comment by fmw42** [comment]
It is still not clear what you want 1080p is a video format. Imagemagick is an image processor.

Nevertheless, a 1080p resolution means the image has 1920 pixels horizontally by 1080 vertically (1920x1080).

So we need to create a black 1920x1080 background and put the text in the center. Since the image is black and text is white, the result is naturally grayscale. If you want color RGB output, then add PNG24: before the output filename with no space.

Using Imagemagick 6, that would be

`convert -size 1920x1080 xc:black -fill white -font Courier -pointsize 72 -gravity center -annotate +0+0 "Text Here" PNG24:output.png`

If you are using Imagemagick 7, then replace convert with magick

**Reply by eddievalentin07** [reply]
Ill try the command. thank you.

**Comment by eddievalentin07** [comment]
> It is still not clear what you want 1080p is a video format. Imagemagick is an image processor.
> 
> Nevertheless, a 1080p resolution means the image has 1920 pixels horizontally by 1080 vertically (1920x1080).
> 
> So we need to create a black 1920x1080 background and put the text in the center. Since the image is black and text is white, the result is naturally grayscale. If you want color RGB output, then add PNG24: before the output filename with no space.
> 
> Using Imagemagick 6, that would be
> 
> `convert -size 1920x1080 xc:black -fill white -font Courier -pointsize 72 -gravity center -annotate +0+0 "Text Here" PNG24:output.png`
> 
> If you are using Imagemagick 7, then replace convert with magick

Is the command that you wrote output lossless and if isn't can you give me another example. I thank you alot and much appreciated.

**Reply by fmw42** [reply, marked as answer]
JPG output is never lossless.  If you want lossless output, then use PNG, or TIFF or some other lossless format.  The best you can do with JPG output is to put -quality 100 before the output (but do not add that for PNG)

So with PNG output as above, the result is lossless.

**Reply by eddievalentin07** [reply]
Thank You So Much you answered my question.
